src/app/integrations/create-page/create-page.component.ts

```diff
@@ -20,27 +20,96 @@ export class IntegrationsCreatePage implements OnInit, OnDestroy {
   integration: Observable<Integration>;
   private readonly loading: Observable<boolean>;
 
-  subscription: Subscription;
+  integrationSubscription: Subscription;
+  routeSubscription: Subscription;
+  childRouteSubscription: Subscription;
+  flowSubscription: Subscription;
+  urls: UrlSegment[];
+  canContinue = false;
+  position: number;
 
-  constructor(private store: IntegrationStore,
+  constructor(
+    private currentFlow: CurrentFlow,
+    private store: IntegrationStore,
     private route: ActivatedRoute,
-    private router: Router) {
+    private router: Router,
+    ) {
     this.integration = this.store.resource;
     this.loading = this.store.loading;
   }
 
+  cancel() {
+    this.router.navigate(['/integrations']);
+  }
+
+  finish() {
+    this.router.navigate(['/integrations']);
+  }
+
+  continue() {
+    const child = this.getCurrentChild();
+    switch (child) {
+      case 'connection-select':
+        this.router.navigate(['connection-configure', this.position], { relativeTo: this.route });
+        break;
+      case 'connection-configure':
+        // TODO hard-coding this to just go to the next connection
+        this.router.navigate(['connection-select', this.position + 1], { relativeTo: this.route });
```

Comment:
Same comment re: URLs. I'd prefer to use location hash to direct/link to place in a flow, just like hash pointing to location in HTML document, but again something we can defer.